extension/mergeSort.js

```diff
@@ -0,0 +1,38 @@
+function mergeSort( initialSet = [] ){
+  if( initialSet == null ){
+    return 'Cannot sort null'
+  }
+  if( initialSet.constructor !== Array ){
+    return 'Please use Arrays'
+  }
+
+  const recurse = (arr) => {
+    let left = arr.slice(0, Math.floor(arr.length / 2))
+    let right = arr.slice(Math.floor(arr.length / 2), arr.length )
+
+    if(left.length > 1){
+      left = recurse(left)
+    }
+    if(right.length > 1){
+      right = recurse(right)
+    }
+
+    let resultArray = []
+    let iIndex = 0
+    let jIndex = 0
+
+    while(iIndex < left.length && jIndex < right.length){
```

Comment:
this is an interesting way to loop through two arrays. Can you accomplish this with nested for loops?